# Start MongoDB database

In principle this step can be avoided if a MongoDB database is available outside the container to read and store data.
I used to have a test database on Mlab and I was able to connect to it from binder.
While this is not working a workaround to have an available mongodb is to install it in the container through conda and start it with a shell command:

`!mongod --dbpath=data/db`

In [ ]:
!mongod --dbpath=data/db

2018-06-14T13:02:35.482+0000 I CONTROL  [initandlisten] MongoDB starting : pid=63 port=27017 dbpath=data/db 64-bit host=jupyter-marcocaggioni-2dmicrorheology-2dtest-2d6tih8nbl
2018-06-14T13:02:35.482+0000 I CONTROL  [initandlisten] db version v3.4.10-4-g67ee356c6b
2018-06-14T13:02:35.482+0000 I CONTROL  [initandlisten] git version: 67ee356c6be377cda547d16423daef5beb4e8377
2018-06-14T13:02:35.482+0000 I CONTROL  [initandlisten] OpenSSL version: OpenSSL 1.0.2o  27 Mar 2018
2018-06-14T13:02:35.482+0000 I CONTROL  [initandlisten] allocator: tcmalloc
2018-06-14T13:02:35.482+0000 I CONTROL  [initandlisten] modules: none
2018-06-14T13:02:35.482+0000 I CONTROL  [initandlisten] build environment:
2018-06-14T13:02:35.482+0000 I CONTROL  [initandlisten]     distarch: x86_64
2018-06-14T13:02:35.482+0000 I CONTROL  [initandlisten]     target_arch: x86_64
2018-06-14T13:02:35.482+0000 I CONTROL  [initandlisten] options: { storage: { dbPath: "data/db" } }
2018-06-14T13:02:35.486+0000 I STORAGE  [inita

2018-06-14T14:13:01.615+0000 I -        [conn9] end connection 127.0.0.1:34704 (4 connections now open)
2018-06-14T14:13:01.615+0000 I -        [conn6] end connection 127.0.0.1:52570 (4 connections now open)
2018-06-14T14:15:01.577+0000 I -        [conn11] end connection 127.0.0.1:54864 (2 connections now open)
2018-06-14T14:15:01.577+0000 I -        [conn10] end connection 127.0.0.1:52714 (2 connections now open)


Once started the database will listen from localhost on the default port 27017 and can be used for the rest of the demonstration